x/staking/keeper/invariants.go

```diff
@@ -65,25 +75,43 @@ func ModuleAccountInvariants(k Keeper) sdk.Invariant {
 
 		k.IterateUnbondingDelegations(ctx, func(_ int64, ubd types.UnbondingDelegation) bool {
 			for _, entry := range ubd.Entries {
-				notBonded = notBonded.Add(entry.Balance.Amount)
+				if val, ok := bonded[entry.Balance.Denom]; ok {
+					notBonded[entry.Balance.Denom] = val.Add(entry.Balance.Amount)
+				} else {
+					notBonded[entry.Balance.Denom]= entry.Balance.Amount
+				}
 			}
 			return false
 		})
 
+		broken := false;
 		poolBonded := k.bankKeeper.GetAllBalances(ctx, bondedPool.GetAddress())
 		poolBondedSum := sdk.ZeroInt()
 		for _, c := range poolBonded {
 			poolBondedSum = poolBondedSum.Add(c.Amount)
+			if val, ok := bonded[c.Denom]; ok {
+				if !val.Equal(c.Amount) {
+					broken = true
```

Comment:
Should I break loop here and next loop? If I break, result of function is exact.